SCons/SConf.py

```diff
@@ -750,11 +751,16 @@ def _startup(self):
                 _ac_config_logs[self.logfile] = None
                 log_mode = "w"
             fp = open(str(self.logfile), log_mode)
+
+            def conflog_cleanup(logf):
+                logf.close()
+
+            atexit.register(conflog_cleanup, fp)
```

Comment:
Can this be done when conf.Finish() is called? or I guess that's not guaranteed to be called?

Reply:
For configure, that's a possibility, if we make sure (as you note) to auto-Finish if it doesn't get called, since there can only be one active logfile for an active Configure. I did these all the same way, registering a call to an inner function with the active file pointer so it's not lost somwhere in the mists.